asserts/ifacedecls_test.go

```diff
@@ -81,24 +81,27 @@ func (s *attrConstraintsSuite) TestSimple(c *C) {
 	cstrs, err := asserts.CompileAttributeConstraints(m["attrs"].(map[string]interface{}))
 	c.Assert(err, IsNil)
 
-	err = cstrs.Check(map[string]interface{}{
+	plug := &snap.PlugInfo{Attrs: map[string]interface{}{
```

Comment:
I'm generally -1 on hand-made `snap.Info` components as they have unrealistic linkage (the snap doesn't know about the plug but the plug knows about the snap`.  We have `snap/snaptest.MockInfo` and some helpers scattered around interface code that extract plug/slot out of that with sane error messages, all designed for testing. Perhaps those should be moved out of their places and into `snaptest` and subsequently used here?